src/test/java/com/rst/bootcamp/InstructorsIntegrationTests.java

```diff
@@ -0,0 +1,142 @@
+package com.rst.bootcamp;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.nio.charset.Charset;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rst.bootcamp.model.Instructor;
+import com.rst.bootcamp.repository.InstructorRepository;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+public class InstructorsIntegrationTests {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	InstructorRepository instructorRepo;
+
+	public static final MediaType APPLICATION_JSON_UTF8 = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Test
+	public void test_addInstructor() throws Exception {
+		Instructor instructor = new Instructor();
+		instructor.setAdminName("A1");
+		instructor.setInstructorName("I1");
+
+		mockMvc.perform(post("/instructor").contentType(APPLICATION_JSON_UTF8).content(asJsonString(instructor)))
+				.andExpect(status().isCreated());
+	}
+
+	
+	public static String asJsonString(final Object obj) {
+		try {
+			return new ObjectMapper().writeValueAsString(obj);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Test
+	public void test_addSubordinate() throws Exception {
```

Comment:
declaration of throws Exception is not required,
Exception is never thrown

Reply:
Understood